interview-questions.md

```diff
@@ -9,59 +9,101 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: "this" refer to the current class'es object.
 
-  Researched answer:
+  Researched answer: The JavaScript this keyword refers to the object it belongs to. It has different values depending on where it is used: In a method, this refers to the owner object. Alone, this refers to the global object. In a function, this refers to the global object.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: JavaScript Lib.
 
-  Researched answer:
+  Researched answer: React is a free and open-source front-end JavaScript library for building user interfaces based on UI components. It is maintained by Meta and a community of individual developers and companies. React can be used as a base in the development of single-page or mobile applications. 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: render()
+
+  Researched answer: The render() method is the only required method in a class component.
+
+When called, it should examine this.props and this.state and return one of the following types:
+
+React elements.        Typically created via JSX. For example, <div /> and <MyComponent /> are React elements that instruct React to render a DOM node, or another user-defined component, respectively.
+Arrays and fragments.  Let you return multiple elements from render. See the documentation on fragments for more details.
+Portals.               Let you render children into a different DOM subtree. See the documentation on portals for more details.
+String and numbers.    These are rendered as text nodes in the DOM.
+Booleans or null.      Render nothing. (Mostly exists to support return test && <Child /> pattern, where test is boolean.)
 
-  Researched answer:
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer:JSX is JavaScript XML, an extension of JavaScript that allows writing HTML in React.
 
-  Researched answer:
+  Researched answer:JSX is JavaScript XML, an extension of JavaScript that allows writing HTML in React. One of the major differences between HTML and JSX is that in JSX, you must return a single parent element, or it won't compile.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Package manager 
 
-  Researched answer:
+  Researched answer: Yarn is a JavaScript package manager that creates the file structure of a React app and manages the dependencies.
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.